File: spring-s3/src/main/java/com/example/springs3/controller/FeedController.java
package com.example.springs3.controller;

import com.example.springs3.controller.dto.CreateFeedRequest;
import com.example.springs3.service.FeedService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.multipart.MultipartFile;

@RestController
@RequiredArgsConstructor
public class FeedController {
    private final FeedService feedService;

    @PostMapping("/post")
    public void post(@RequestBody CreateFeedRequest request) {
        feedService.uploadFeed(request);
    }
}

File: spring-kotlin/src/main/kotlin/com/example/springkotlin/global/feed/presentation/FeedController.kt
package com.example.springkotlin.global.feed.presentation

import com.example.springkotlin.global.feed.presentation.dto.CreateFeedRequest
import com.example.springkotlin.global.feed.service.CreateFeedService
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/feed")
class FeedController(
    private val createFeedService: CreateFeedService
) {

    @ResponseStatus(HttpStatus.CREATED)
    @PostMapping
    fun createFeed(@RequestBody request: CreateFeedRequest) {
        createFeedService.execute(request)
    }
}